core/src/main/scala/io/qbeast/core/model/CubeWeights.scala

```diff
@@ -144,9 +199,19 @@ private class WeightAndCount(var weight: Weight, var count: Int)
 protected case class PointWeightAndParent(point: Point, weight: Weight, parent: Option[CubeId])
 
 /**
- * Cube and NormalizedWeight
- *
- * @param cubeBytes        the cube
- * @param normalizedWeight the weight
+ * NormalizedWeight and tree size of a given cube, with tree size defined as the
+ * sum of its own payload and all that of its descendents. The tree size is
+ * initialized with cube size and updated to the tree size during computation
+ * before being used.
+ * @param weight NormalizedWeight
+ * @param treeSize Cube tree size
+ */
+protected class WeightAndTreeSize(val weight: NormalizedWeight, var treeSize: Double)
```

Comment:
Could you review the use of `var` in this class @alexeiakimov ? Do you think is safe?

Reply:
I think it is safe if it is a class used only internally. However, why is it protected and not private? I understand **protected** allows access to subclasses. Is that case? Is there any subclass accessing this method? If not, it is better to put it private, right?

Reply:
By the way, we can replace all protected with private in this class.

Reply:
I'll change all protected variables and classes to private.